src/north_mcp_python_sdk/auth.py

```diff
@@ -80,30 +90,56 @@ class NorthAuthBackend(AuthenticationBackend):
     Authentication backend that validates Bearer tokens.
     """
 
-    def __init__(self, server_secret: str | None = None):
+    def __init__(self, server_secret: str | None = None, debug: bool = False):
         self._server_secret = server_secret
+        self.debug = debug
+        self.logger = logging.getLogger("NorthMCP.Auth")
+        if debug:
+            self.logger.setLevel(logging.DEBUG)
 
     async def authenticate(
         self, conn: HTTPConnection
     ) -> tuple[AuthCredentials, BaseUser] | None:
+        if self.debug:
+            self.logger.debug(f"Authenticating request from {conn.client}")
```

Comment:
I think you should avoid interpolating strings in logs. Just pass the variables as context to the logger instead